python/ray/_raylet.pyx

```diff
@@ -941,10 +939,12 @@ cdef class CoreWorker:
             message.decode("utf-8")))
 
     def submit_task(self,
+                    Language language,
                     function_descriptor,
                     args,
                     int num_return_vals,
                     c_bool is_direct_call,
+                    c_bool is_cross_language,
```

Comment:
What's the use of this flag? It seems to me the caller can always check if it's cross language by seeing if the task language != the current language.

Maybe we could instead include the caller language in addition to the task language?

Reply:
Currently, this flag is useless for TaskSpec, I will remove this flag from TaskSpec in next commit. But, is_cross_language flag is useful for cross-lang serialization, so I passed it to cython for cross-lang serialization feature.

Src(caller) language & dst(callee) language is enough for cross different languages, but is_cross_language can implement same language cross-lang call, e.g. Python call Python, which is useful for remote client - a lightwight client connect to the running job can call job APIs without source code. Also, Python call Python is convenient for dev.

Reply:
It sounds like the flag should be called `use_xlang_serialization` or something more precise?

Reply:
is_cross_language is not only for object serialization but also for submit task logic.

Reply:
Would you mind add some comments in the code for the difference between is_cross_language and language?

Reply:
I'd like to rename the 1st param `language` to `target_language` and rename the last param `is_cross_language` to `source_language` instead.
ps: Some explicit comments for them are necessary.